### ISSUE TITLE
ISI crash on Reload arrangement

### ISSUE THREAD

**Issue by jens-maus** [jens-maus/yam#413]
**Originally by _s.hawamdeh@teletu.it_ on 2013-08-26 09:59:28 +0200**

___
The reload arrangement (accessible via right click on the folder list column at the left) seems a bit stable from time to time, i already experience some related but not so problematic DSI, but now i had also an ISO one

What is strange is that the issue is not always reproducible but it happen also in a perfecly clean YAM installation (aka no mail at all received nor sended)

**Comment by jens-maus** [comment]
**Originally by _s.hawamdeh@teletu.it_ on 2013-08-26 09:59:57 +0200**

___
**Attachment added:** [`Crashlog_YAM_2013-08-26_05-23-09.txt`](https://yam.ch/attachments/413/Crashlog_YAM_2013-08-26_05-23-09.txt) (30.7 KiB)
ISI crash

**Comment by jens-maus** [comment]
**Originally on 2013-08-26 12:53:16 +0200**

___
Please note, that with ticket #284 this ticket will be automatically solved as the reload functionality will be removed then,

**Comment by jens-maus** [comment]
**Originally by _s.hawamdeh@teletu.it_ on 2013-08-26 16:45:05 +0200**

___
@damato

Ok fine, however now i'm not 100% sure if the problem are really related to the arrangement functionality, some days ago i got a similar issue but just selecting one of the folder .. (i already reported the issue to Thore)

Maybe isn't related at all, or maybe is related ?
Need to verify (but first i need to reproduce the error)

**Comment by tboeckel** [comment]
**Originally on 2013-08-27 21:05:09 +0200**

___
So far I was not able to reproduce this issue myself. Maybe you were a victim of trashed memory caused by other applications?

**Comment by jens-maus** [comment]
**Originally by _s.hawamdeh@teletu.it_ on 2013-09-02 07:30:34 +0200**

___
@Thore

I don't think so as i was able to reproduce the crash only having YAM opened and nothing else, however the latest nighly builds does not start anymore on my system (see new ticket 416 for more info) so i wasn't able to test if the latest nightly fixed it

So i downloaded and tested YAM 2.8p1 and the stable version does not produce any crash reloading the arrangement, the crash on the reload arrangement might be an issue added in recent nightly build (?) ..

**Comment by tboeckel** [comment]
**Originally on 2013-09-09 14:08:00 +0200**

___
What about reproducing this issue now after #416 is fixed?

**Comment by jens-maus** [comment]
**Originally by _s.hawamdeh@teletu.it_ on 2013-09-13 13:03:41 +0200**

___
**Attachment added:** [`Crashlog_YAM_2013-09-10_10-01-29.txt`](https://yam.ch/attachments/413/Crashlog_YAM_2013-09-10_10-01-29.txt) (31.8 KiB)
DSI crash

**Comment by jens-maus** [comment]
**Originally by _s.hawamdeh@teletu.it_ on 2013-09-13 13:04:34 +0200**

___
**Attachment added:** `no_icon_config.png` (648.0 KiB)
![no_icon_config.png](https://yam.ch/attachments/413/no_icon_config.png)
No graphics in prefs window

**Comment by jens-maus** [comment]
**Originally by _s.hawamdeh@teletu.it_ on 2013-09-13 13:06:17 +0200**

___
Here is me again :-)
Sorry for delay, it was a busy week !

I just retried with the very latest nightly build (2013-09-13) and i was NOT able to reproduce the issue anymore.
Of course i'm not sure if the issue is really fixed in meantime, so i'm doing some more test ..

However atleast with an old nightly (2013-09-09 for example) i had a DSI crash when i try to reload the arrangement!

See log attached 

To reproduce it i just reinstalled YAM 2.8p1 in a clean partition and then i've updated it to a recent 2.9 nightly.

The config was of course empty (no configured), so no mail sent/received at all, aka my YAM installation was and still completely clean

Aniway above this specific issue i also noted that at first start the preferences window of YAM does not show the icons (see grab)

Instead if i close the prefs window and then i reopen it again the icon will be showed normally, please can you fix it aswell ?

**Comment by tboeckel** [comment]
**Originally on 2013-09-13 14:07:14 +0200**

___
In (fc8d7c5):
 * YAM.c: moved the validation of the config after loading the configured theme. This solves the problem when YAM is run for the very first time after an installation from scratch without any valid .config file. Opening the config window too early will leave this window without any graphics from the default theme otherwise. This refs #413.

**Comment by tboeckel** [comment]
**Originally on 2013-09-13 14:13:37 +0200**

___
Replying to samo79:

> I just retried with the very latest nightly build (2013-09-13) and i was NOT able to reproduce the issue anymore.
> Of course i'm not sure if the issue is really fixed in meantime, so i'm doing some more test ..

Ok, I am going to close this ticket as "fixed" in the meantime. If the crash should happen again then please reopen this ticket.

> Aniway above this specific issue i also noted that at first start the preferences window of YAM does not show the icons (see grab)

This is fixed with my latest change.
 
> Instead if i close the prefs window and then i reopen it again the icon will be showed normally, please can you fix it aswell ?

I did. But please open a new ticket for such an issue next time, because there is absolutely no relationship between a crash after a certain action and missing icons. Mixing several unrelated bugs in a single ticket might result in bugs to be forgotten.

**Comment by jens-maus** [comment]
**Originally by _s.hawamdeh@teletu.it_ on 2013-09-13 14:18:24 +0200**

___
Ok thanks for the theme fix, will try the new build tonight and eventually, if still i will open another ticket for the reload arrangement issue :-)

**Comment by tboeckel** [comment]
**Originally on 2013-09-13 14:26:07 +0200**

___
Replying to samo79:
> Ok thanks for the theme fix, will try the new build tonight and eventually, if still i will open another ticket for the reload arrangement issue :-)

Sorry, but you still got me wrong. For the "reload arrangement issue" you should reopen **THIS** ticket, because this ticket already is about the "reload arrangement issue". However, for any other issue whatsoever please open a **NEW** ticket.

**Comment by jens-maus** [comment]
**Originally by _s.hawamdeh@teletu.it_ on 2013-09-13 14:40:10 +0200**

___
Ah ops !

**Comment by jens-maus** [comment]
**Originally by _s.hawamdeh@teletu.it_ on 2013-09-17 23:05:53 +0200**

___
**Attachment added:** [`Crashlog_YAM_2013-09-17_21-01-04.txt`](https://yam.ch/attachments/413/Crashlog_YAM_2013-09-17_21-01-04.txt) (28.9 KiB)
Reload arrangement crash

**Comment by jens-maus** [comment]
**Originally by _s.hawamdeh@teletu.it_ on 2013-09-17 23:06:26 +0200**

___
Ok i was able to reproduce the issue, however reproduce it still rather difficul (casual ??), because it is presented in a inconsistent manner (sometimes it crash and sometimes not for no apparent reason!)

The only thing I've realized is that you do not need to do a Reload arrangement in a specific folder, even click on a column can be enough

So right click with the mouse -> Reload arrangment

Repeat the operation some times again and again and after a while you should have the crash !

**Comment by tboeckel** [comment]
**Originally on 2013-09-18 08:57:10 +0200**

___
I reloaded my folder tree about a 100 times by now, but for me there is no crash or memory corruption at all. Please try again with the next nightly build, as I just changed a minor thing abd used that version for the test, which might have an impact on this issue.

**Comment by jens-maus** [comment]
**Originally by _s.hawamdeh@teletu.it_ on 2013-09-20 14:06:56 +0200**

___
Problem still also with the latest nightly build (2013-09-20) however the crash seems less reproducible now 

As before (as test) i reloaded the folder tree many time but i got a crash only when i reopen YAM after an iconification --> (i was using the system with other program in background)

**Comment by jens-maus** [comment]
**Originally by _s.hawamdeh@teletu.it_ on 2013-09-20 14:07:42 +0200**

___
**Attachment added:** [`Crashlog_YAM_2013-09-20_06-06-41.txt`](https://yam.ch/attachments/413/Crashlog_YAM_2013-09-20_06-06-41.txt) (28.1 KiB)
Crash on nightly build (2013-09-20)

**Comment by tboeckel** [comment]
**Originally on 2013-09-23 20:16:10 +0200**

___
Replying to samo79:
> Problem still also with the latest nightly build (2013-09-20) however the crash seems less reproducible now 
> 
> As before (as test) i reloaded the folder tree many time but i got a crash only when i reopen YAM after an iconification --> (i was using the system with other program in background)

Sorry, but I am still not able to reproduce this. No matter how often I reload the folder tree and/or (de)iconify YAM everything is absolutely stable.

The strange thing about your latest crashlog is the fact that the crash happens in 68k code instead of PPC code. There is really something fishy happening on your side.

**Comment by jens-maus** [comment]
**Originally by _s.hawamdeh@teletu.it_ on 2013-09-25 00:12:18 +0200**

___
Strange, all can i do at this point is retry with a clean OS installation... in that case i think we should wait some time for that test  --> aka when i need to reinstall the OS --> not so far :-)

**Comment by jens-maus** [comment]
**Originally on 2013-12-22 23:12:51 +0100**

___
**Release Note:**
fixed a problem when YAM was run for the very first time after an installation without any valid .config file. Opening the config window too early caused this window to be opened without any graphics from the default theme.
